django_socio_grpc/proto_serializers.py

```diff
@@ -76,59 +85,59 @@ def populate_dict_with_none_if_not_required(self, data_dict, message=None):
-                PARTIAL_UPDATE_FIELD_NAME, {}
-            ):
-                if field.field_name in data_dict:
-                    del data_dict[field.field_name]
+            # INFO - AM - 04/01/2024 - If we are in a partial serializer we only
+            # need to have field specified in PARTIAL_UPDATE_FIELD_NAME attribute
+            # in the data. Meaning deleting fields that should not be here and not adding
+            # None to allow_null field that are not specified
+            if self.partial and field.field_name not in partial_fields:
                 continue
-            # INFO - AM - 04/01/2024 - if field already existing in the data_dict we do not need to do something else
+
             if field.field_name in data_dict:
+                clean_dict[field.field_name] = data_dict[field.field_name]
                 continue
 
-            # INFO - AM - 04/01/2024 - if field is not in the data_dict but in PARTIAL_UPDATE_FIELD_NAME we need to set the default value if existing or raise exception to avoid having default grpc value by mistake
-            if self.partial and field.field_name in data_dict.get(
-                PARTIAL_UPDATE_FIELD_NAME, {}
-            ):
-                if field.allow_null:
-                    data_dict[field.field_name] = None
-                    continue
-                if field.default not in [None, empty]:
-                    data_dict[field.field_name] = get_default_value(field.default)
-                    continue
-
-                # INFO - AM - 11/03/2024 - Here we set the default value especially for the blank authorized data. We debated about raising a ValidaitonError but prefered this behavior. Can be changed if it create issue with users
-                data_dict[field.field_name] = message.DESCRIPTOR.fields_by_name[
-                    field.field_name
-                ].default_value
-
-            if field.allow_null or (field.default in [None, empty] and field.required is True):
-                if is_update_process:
-                    data_dict[field.field_name] = None
-                    continue
-
-                if field.default not in [None, empty]:
-                    data_dict[field.field_name] = None
-                    continue
-
-                if (
-                    hasattr(self, "Meta")
-                    and hasattr(self.Meta, "model")
-                    and hasattr(self.Meta.model, field.field_name)
-                ):
-                    deferred_attribute = getattr(self.Meta.model, field.field_name)
-                    if deferred_attribute.field.default != NOT_PROVIDED:
-                        data_dict[field.field_name] = get_default_value(
-                            deferred_attribute.field.default
-                        )
-                        continue
-
-                data_dict[field.field_name] = None
-        return data_dict
+            with contextlib.suppress(_NoDictData):
```

Comment:
can you add a comment on that line to explain what it does but mostly why do we need it this way  ?

Reply:
This is from ruff